web/src/app/services/IntegrationKeyForm.tsx

```diff
@@ -13,11 +12,29 @@ import { AppLink } from '../util/AppLink'
 
 const useStyles = makeStyles((theme) => ({
   infoIcon: {
-    color: theme.palette.primary['500'],
+    color: theme.palette.primary.main,
   },
 }))
 
-export default function IntegrationKeyForm(props) {
+interface Value {
+  name: string
+  type: 'generic' | 'grafana' | 'site24x7' | 'email'
+}
+
+interface IntegrationKeyFormProps {
+  value: Value
+
+  errors: {
+    field: ['name' | 'type']
```

Comment:
Shouldn't be an array:

```suggestion
    field: 'name' | 'type'
```

Reply:
Made requested changes.